### ISSUE TITLE
Magento crashes after product upload

### ISSUE THREAD

**Issue by miketsl** [magento/magento2#4829]
## Steps to reproduce
1. Install Magento from `develop` branch.
2. Add Simple products to Catalog, using size and color eav
3. Add configurable product linked to the simple products just added
4. Go to store homepage, or the configurable product page directly
## Expected result
1. Store functions as expected
## Actual result
1. Homepage just shows the picture of the configurable product, with a link to it, rest of homepage missing.
2. Product page (if you click link) shows white blank page.
3. httpd error log reports:
   Call to a member function getValue() on a non-object in /var/www/html/user049/new/magento2/vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php on line 101
4. code from FinalPriceBox.php on line 101 is:
   public function hasSpecialPrice()
     {
         $displayRegularPrice = $this->getPriceType(Price\RegularPrice::PRICE_CODE)->getAmount()->getValue();
         $displayFinalPrice = $this->getPriceType(Price\FinalPrice::PRICE_CODE)->getAmount()->getValue();
         return $displayFinalPrice < $displayRegularPrice;
     }
5. Screenshots:
   ![image](https://cloud.githubusercontent.com/assets/16725994/15740277/cb3a2b3c-28e6-11e6-9493-6aa53ca9617c.png)
   ![image](https://cloud.githubusercontent.com/assets/16725994/15740295/e0aae3ee-28e6-11e6-8e12-eae0cf265203.png)

**Comment by miketsl** [comment]
BTW, its magento 2.0.7...

**Comment by miketsl** [comment]
Solved this problem.  
This strange behavior occurs when you mass import products using "Default" category in Magento 2.0.7.  In the Magento database, there are actually something like 7 attribute sets all named "Default".  Since I put "Default" in the csv file, I am assuming it chose one randomly, which was attribute set id 4 in my case, and lead to all sorts of weird Magento behavior.

To solve I created a new attribute set, and then changed my spreadsheet to reflect this set.
